webapp/security/views.py

```diff
@@ -268,8 +268,64 @@ def api_create_notice():
 # ===
 def cve_index():
 
-    list_cve = db_session.query(CVE).limit(10)
-    return flask.render_template("security/cve/index.html", list_cve=list_cve)
+    # Query parameters
+    order_by = flask.request.args.get("order_by", default="oldest", type=str)
+    search_desc = flask.request.args.get("search_desc", default="", type=str)
+    package = flask.request.args.get("package", default="", type=str)
+    priority = flask.request.args.get("priority", default="", type=str)
+    limit = flask.request.args.get("limit", default=20, type=int)
+    offset = flask.request.args.get("offset", default=0, type=int)
+
+    # Basic queries
+    cves_query = db_session.query(CVE)
+    releases_query = db_session.query(Release)
+
+    # Search functionality
+    if search_desc and priority:
+        cves_query = cves_query.filter(
+            and_(
+                CVE.description.ilike(f"%{search_desc}%"),
+                func.lower(CVE.priority) == func.lower(priority),
+            )
+        )
+    elif priority:
+        cves_query = cves_query.filter(CVE.priority == priority,)
+    elif search_desc:
+        cves_query = cves_query.filter(
+            CVE.description.ilike(f"%{search_desc}%"),
+        )
+
+    # Pagination
+    total_results = cves_query.count()
+    total_pages = ceil(total_results / limit)
+
+    if offset > total_pages:
+        flask.abort(404)
```

Comment:
Seeing a 404 page in this case is very surprising. If the offset is too large, please just return an empty list.